### ISSUE TITLE
Build instructions for MOTION2NX

### ISSUE THREAD

**Issue by goxul** [encryptogroup/MOTION2NX#1]
Hello,

Are there any build instructions available, on how to get this running? 

I followed the same instructions that were present in the MOTION repository - `mkdir build && cd build`, followed by `cmake ..` and `make`. 

The first two steps worked fine - here's the output of `cmake ..`:

```
-- The CXX compiler identification is GNU 11.1.0
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for C++ include pthread.h
-- Looking for C++ include pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Found OpenSSL: /usr/lib/x86_64-linux-gnu/libcrypto.so (found version "1.1.1f")  
-- Found OpenMP_CXX: -fopenmp (found version "4.5") 
-- Found OpenMP: TRUE (found version "4.5")  
-- Found Boost: /usr/lib/x86_64-linux-gnu/cmake/Boost-1.71.0/BoostConfig.cmake (found suitable version "1.71.0", minimum required is "1.69.0") found components: context fiber filesystem log log_setup program_options system thread 
fmt was not found: add fmt subdirectory
flatbuffers was not found: add flatbuffers subdirectory
-- The C compiler identification is GNU 11.1.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Looking for strtof_l
-- Looking for strtof_l - found
-- Looking for strtoull_l
-- Looking for strtoull_l - found
-- Configuring done
-- Generating done
-- Build files have been written to: /home/gokul/MOTION2NX/build
```

However, `make` fails for some reason - here's the error log [attached](https://indianinstituteofscience-my.sharepoint.com/:t:/g/personal/gokulnathp_iisc_ac_in/EYS66kUtYaBJmYvU9B-0QMMBsiDgS4TIi5SMq0ZUDCP2kg?e=Fd0y8z)

I was able to build MOTION without any problems by installing the mentioned pre-requisites, so I thought the same would work here. Is there any dependency that I am missing?

Here are some of the relevant specs, please let me know if you need anything else:

```
gokul@cssl:~/MOTION2NX/build$ g++ --version
g++ (Ubuntu 11.1.0-1ubuntu1~20.04) 11.1.0
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

gokul@cssl:~/MOTION2NX/build$ cmake --version
cmake version 3.16.3

CMake suite maintained and supported by Kitware (kitware.com/cmake).
gokul@cssl:~/MOTION2NX/build$ make --version
GNU Make 4.2.1
Built for x86_64-pc-linux-gnu
Copyright (C) 1988-2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

**Comment by lenerd** [comment]
Hi,
thanks for your interest! I can't say off top of my head what is going wrong there, but writing proper build instructions are definitely on my TODO list.

**Comment by goxul** [comment]
Thanks for the prompt response -  I wasn't running Make with C++11 support before, post that I was able to move past that error. Once I changed my CXX and CXXFLAGS, I was able to get ride of most of the errors.

I was able to finally build, but I had to edit quite a bit - add a few headers files, change some header files to point to different locations etc. Boost 1.71 and 1.77 didn't work, but surprisingly 1.76 worked.

I built this on Ubuntu 20.04.

**Comment by goxul** [comment]
If you can, could you please let me know what version of Boost are you using where this works? I am having trouble with a particular header file - the code here uses `#include<boost/json/to_string.hpp>` whereas none of the modern releases of the JSON library have this header file in this location that you've mentioned. 

Could you kindly confirm once?

**Comment by lenerd** [comment]
I added build instructions to the README and fixed some things that I found when enabling C++20 and compiling the whole thing for the first time in a year with newer compiler and library versions.

Note that the stated version numbers refer to the versions that I used for testing. Hence, it might very well be the case that older versions of some libraries suffice.

Regarding Boost.JSON: I started using the library while it was still under review and not yet integrated into Boost. The API has changed a bit since then. This is fixed now.
